storage/watchable_store.go

```diff
@@ -356,3 +360,25 @@ type watching struct {
 	// The chan might be shared with other watchings.
 	ch chan<- storagepb.Event
 }
+
+// safelySynced safely puts watching with key k into watchableStore's synced.
+func safelySynced(s *watchableStore, k string, wa *watching) error {
```

Comment:
actually i think this should be names as Unsafe since you need to hold the lock before doing it, right?